datastore/sampling/cross_validation.py

```diff
@@ -0,0 +1,34 @@
+from collections import namedtuple
+from sklearn.model_selection import StratifiedKFold
+from datastore.api.data import Subset
+
+
+def stratified_split(dataset, num_splits):
+    """ Create stratified k-fold splits
+
+    Parameters
+    ----------
+    dataset : datastore.dataset
+
+    num_splits : int
+
+    Returns
+    -------
+    splits : list(namedtuples<subset, subset>)
+        stratified splits of the data
+    """
+    skf = StratifiedKFold(n_splits=num_splits)
+    data, labels = dataset.load_data()
```

Comment:
This requires that datasets be able to load all the data in memory. It also forces that these in memory datasets to have a method called `load_data()` that returns the data and the labels.